core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1869,8 +1869,7 @@ public void _run() throws IOException, InstallationStatus {
                 // Do this first so we can avoid duplicate downloads, too
                 // check to see if the plugin is already installed at the same version and skip it
                 LOGGER.info("Skipping duplicate install of: " + plugin.getDisplayName() + "@" + plugin.version);
-                //throw new Skipped(); // TODO set skipped once we have a status indicator for it
-                return;
+                throw new Skipped();
```

Comment:
@daniel-beck, if this is cause by this PR, it must be this. Seems the comment was/got wrong with time. We should probably stick to `return` and remove the comment since it worked well in the past.